jupyter/d2l-java/chapter_linear-networks/linear-regression.ipynb

```diff
@@ -0,0 +1,775 @@
+    "\n",
+    "$${\\hat{\\mathbf{y}}} = \\mathbf{X} \\mathbf{w} + b.$$\n",
+    "\n",
+    "Given a training dataset $\\mathbf{X}$\n",
+    "and corresponding (known) targets $\\mathbf{y}$,\n",
+    "the goal of linear regression is to find\n",
+    "the *weight* vector $w$ and bias term $b$\n",
+    "that given a new data point $\\mathbf{x}_i$,\n",
+    "sampled from the same distribution as the training data\n",
+    "will (in expectation) predict the target $y_i$ with the lowest error.\n",
+    "\n",
+    "Even if we believe that the best model for\n",
+    "predicting $y$ given $\\mathbf{x}$ is linear,\n",
+    "we would not expect to find real-world data where\n",
+    "$y_i$ exactly equals $\\mathbf{w}^\\top \\mathbf{x}+b$\n",
+    "for all points ($\\mathbf{x}, y)$.\n",
+    "For example, whatever instruments we use to observe\n",
+    "the features $\\mathbf{X}$ and labels $\\mathbf{y}$\n",
+    "might suffer small amount of measurement error.\n",
+    "Thus, even when we are confident\n",
+    "that the underlying relationship is linear,\n",
+    "we will incorporate a noise term to account for such errors.\n",
+    "\n",
+    "Before we can go about searching for the best parameters $\\mathbf{w}$ and $b$,\n",
+    "we will need two more things:\n",
+    "(i) a quality measure for some given model;\n",
+    "and (ii) a procedure for updating the model to improve its quality.\n",
+    "\n",
+    "### Loss Function\n",
+    "\n",
+    "Before we start thinking about how *to fit* our model,\n",
+    "we need to determine a measure of *fitness*.\n",
+    "The *loss function* quantifies the distance\n",
+    "between the *real* and *predicted* value of the target.\n",
+    "The loss will usually be a non-negative number\n",
+    "where smaller values are better\n",
+    "and perfect predictions incur a loss of $0$.\n",
+    "The most popular loss function in regression problems\n",
+    "is the sum of squared errors.\n",
+    "When our prediction for an example $i$ is $\\hat{y}^{(i)}$\n",
+    "and the corresponding true label is $y^{(i)}$,\n",
+    "the squared error is given by:\n",
+    "\n",
+    "$$l^{(i)}(\\mathbf{w}, b) = \\frac{1}{2} \\left(\\hat{y}^{(i)} - y^{(i)}\\right)^2.$$\n",
+    "\n",
+    "The constant $1/2$ makes no real difference\n",
+    "but will prove notationally convenient,\n",
+    "cancelling out when we take the derivative of the loss.\n",
+    "Since the training dataset is given to us, and thus out of our control,\n",
+    "the empirical error is only a function of the model parameters.\n",
+    "To make things more concrete, consider the example below\n",
+    "where we plot a regression problem for a one-dimensional case\n",
+    "as shown in :numref:`fig_fit_linreg`.\n",
+    "\n",
+    "![Fit data with a linear model.](https://raw.githubusercontent.com/awslabs/djl-resources/311169c9dbd89e1597a5ae6b0b1ba8a402b8b55e/jupyter/d2l-java/img/Neuron.svg)\n",
+    ":label:`fig_fit_linreg`\n",
```

Comment:
this doesn't render properly, any idea? d2l seems working fine, is there any plugin we missed?

Reply:
we can embed url links with html
so instead of ![alt text](local img path) we can do
<img src="https://raw.githubusercontent.com/awslabs/djl-resources/311169c9dbd89e1597a5ae6b0b1ba8a402b8b55e/jupyter/d2l-java/img/Neuron.svg"/>

Reply:
oh that works in here as well LOL
`<img src="https://raw.githubusercontent.com/awslabs/djl-resources/311169c9dbd89e1597a5ae6b0b1ba8a402b8b55e/jupyter/d2l-java/img/Neuron.svg"/>`